File: worker/src/main/java/com/github/k0kubun/gitstar_ranking/repository/dao/LockDao.java
package com.github.k0kubun.gitstar_ranking.repository.dao;

import org.skife.jdbi.v2.sqlobject.Bind;
import org.skife.jdbi.v2.sqlobject.SqlUpdate;

public interface LockDao
{
    @SqlUpdate("select pg_advisory_xact_lock(:key)")
    void getLock(@Bind("key") long key);
}

File: worker/src/main/kotlin/com/github/k0kubun/gitstar_ranking/repository/dao/LockDao.kt
package com.github.k0kubun.gitstar_ranking.repository.dao

import org.skife.jdbi.v2.sqlobject.SqlUpdate
import org.skife.jdbi.v2.sqlobject.Bind

interface LockDao {
    @SqlUpdate("select pg_advisory_xact_lock(:key)")
    fun getLock(@Bind("key") key: Long)
}
